spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/metrics/MetricsUpdater.scala

```diff
@@ -0,0 +1,14 @@
+package com.datastax.spark.connector.metrics
+
+import org.apache.spark.SparkEnv
+
+trait MetricsUpdater {
+  def finish(): Long
+
+  def forceReport(): Unit = {
+    val env = SparkEnv.get
```

Comment:
well... everyone is still learning. presumably in each project you can find better and worse parts of the code